go/test/endtoend/vtgate/misc_test.go

```diff
@@ -780,12 +781,56 @@ func TestJoinWithMergedRouteWithPredicate(t *testing.T) {
 }
 
 func TestRowCountExceed(t *testing.T) {
-	conn, closer := start(t)
-	defer closer()
+	conn, _ := start(t)
+	defer func() {
+		cluster.PanicHandler(t)
+		// needs special delete logic as it exceeds row count.
+		for i := 50; i <= 300; i += 50 {
+			utils.Exec(t, conn, fmt.Sprintf("delete from t1 where id1 < %d", i))
+		}
+		conn.Close()
+	}()
 
 	for i := 0; i < 250; i++ {
 		utils.Exec(t, conn, fmt.Sprintf("insert into t1 (id1, id2) values (%d, %d)", i, i+1))
 	}
 
 	utils.AssertContainsError(t, conn, "select id1 from t1 where id1 < 1000", `Row count exceeded 100`)
 }
+
+func TestLookupErrorMetric(t *testing.T) {
+	conn, closer := start(t)
+	defer closer()
+
+	var errCount float64
+	apiErr := getVar(t, "VtgateApiErrorCounts")
+	if apiErr != nil {
+		mapErrors := apiErr.(map[string]interface{})
+		val, exists := mapErrors["Execute.ks.primary.ALREADY_EXISTS"]
+		if exists {
+			errCount = val.(float64)
+		}
+	}
+
+	utils.Exec(t, conn, `insert into t1 values (1,1)`)
+	_, err := utils.ExecAllowError(t, conn, `insert into t1 values (2,1)`)
+	require.ErrorContains(t, err, `(errno 1062) (sqlstate 23000)`)
+
+	apiErr = getVar(t, "VtgateApiErrorCounts")
+	require.NotNil(t, apiErr)
+	mapErrors := apiErr.(map[string]interface{})
+	val, exists := mapErrors["Execute.ks.primary.ALREADY_EXISTS"]
+	require.True(t, exists)
+	require.EqualValues(t, errCount+1, val)
+}
+
+func getVar(t *testing.T, key string) interface{} {
```

Comment:
in both places where you are using the output of this method, you cast it to a `map[string]interface{}`. Why not have this method return that type instead of `interface{}`?

Also, I think these days we want to use `any` instead of `interface{}`